src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerImpl.scala

```diff
@@ -28,11 +28,103 @@ import scala.concurrent.{ Await, ExecutionContext, Future }
 import scala.util.control.NonFatal
 import scala.util.{ Failure, Success }
 
+trait AssignDynamicServiceLogic extends StrictLogging {
```

Comment:
Why do we need an extra trait just to cover this method?

Reply:
+1

Reply:
It's easier to test this way. We don't need mocked repositories etc for testing this. I'd love a better pattern but cannot think of one.

Reply:
No. It was just  a fast fix. I merged your patch.